modules/messages/server/jobs/message-unread.server.job.js

```diff
@@ -21,27 +21,81 @@ var _ = require('lodash'),
     path = require('path'),
     emailService = require(path.resolve('./modules/core/server/services/email.server.service')),
     log = require(path.resolve('./config/lib/logger')),
+    config = require(path.resolve('./config/config')),
     async = require('async'),
     moment = require('moment'),
     mongoose = require('mongoose'),
     Message = mongoose.model('Message'),
     User = mongoose.model('User');
 
 module.exports = function(job, agendaDone) {
+
+  // here we expect an array of momentjs objects
+  // i.e. [{ minutes: 10 }, { hours: 24 }]
+  var remindersConfig = config.limits.unreadMessageReminders;
+
+  // sort the config from the earliest to the latest
+  var sortedConfig = _.sortBy(remindersConfig, function (timeSinceMessage) {
+    return moment.duration(timeSinceMessage).asMilliseconds();
+  });
+
+  /* we run the reminders from the last to the first.
+   * that way we avoid duplicates when more reminders are pending
+   * i.e. when notificationCount is 0 and 24 hours already passed (maybe due to some agenda outage)
+   * we run the 2nd reminder first, the notificationCount is set to 2
+   * and the 1st reminder is not sent at all
+   *
+   * otherwise both reminders would be sent at the same time o_O
+   *
+   * hopefully there is not much harm in sending just one reminder in total
+   * this case shouldn't happen too often
+   */
+
+  var remappedConfig = _.reverse(_.map(sortedConfig, function (value, index) {
+    return {
+      order: index, // nth notification
+      delay: value // when to send the notification
+    };
+  }));
+
+  // TODO check that the config has a proper format
+
+  // sendUnreadMessageReminders(remindersConfig[0], 0, agendaDone);
+  async.eachSeries(remappedConfig, sendUnreadMessageReminders, agendaDone);
+};
+
+/**
+ * // send the reminders about unread messages to various services
+ * @param {Object} reminder - info about the reminder
+ * @param {Object} reminder.delay - a time object as momentjs would expect it, i.e. { 'minutes': 10 }
+ * @param {number} reminder.order - which reminder are we sending: i.e. 0 for the first reminder, 5 for the sixth reminder
+ * @param {Function} callback - a callback function
+ */
+function sendUnreadMessageReminders(reminder, callback) {
+
+  var timePassed = reminder.delay;
+  var reminderPosition = reminder.order;
+
   async.waterfall([
 
     // Aggregate unread messages
     function(done) {
 
-      // Ignore very recent messages and look for only older than 10 minutes
+      // We want to remind user about messages, which remain unread for more than `timePassed`
+      // But only once for each `timePassed`
       // Has to be a JS Date object, not a Moment object
-      var createdTimeAgo = moment().subtract(moment.duration({ 'minutes': 10 })).toDate();
+      var createdTimeAgo = moment().subtract(moment.duration(timePassed)).toDate();
 
+      //
       Message.aggregate([
```

Comment:
This code finds all the unread, unnotified messages older than a specified period of time, groups them together by sender+receiver and sends notifications.

If i send 4 messages in 5-minute intervals, it seems to me that 4 separate notifications will be sent in 5-minute intervals starting 10 - 15 minutes later. (Only less than 5 minute messages will be sometimes grouped, because agenda runs every 5 minutes). That seems to be the original behaviour. Is it intended?

If that's true the same would happen 24 hours later with 2nd notifications. In that case i suggest to wait until 24 hours pass since the very last unread not-2nd-time-notified message, and then to send one reminder with all the unnotified unread messages included. Does this make sense?

@simison @chmac @nicksellen how do you imagine the desired behavior of notifications, and 2nd notifications?